tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -116,7 +116,11 @@ internal TracerSettings(IConfigurationSource? source, IConfigurationTelemetry te
                                 _ when x.ToBoolean() is { } boolean => boolean,
                                 _ => ParsingResult<bool>.Failure(),
                             },
-                            getDefaultValue: () => false,
+                            getDefaultValue: () =>
+                            {
+                                var profilingSsiDeployed = config.WithKeys(ContinuousProfiler.ConfigurationKeys.SsiDeployed).AsString();
+                                return (profilingSsiDeployed != null) ? profilingSsiDeployed.Contains("profiler") : false;
+                            },
```

Comment:
nit: I would probably simplify this a little
```suggestion
                            config
                               
 .WithKeys(ContinuousProfiler.ConfigurationKeys.SsiDeployed)
                                .AsString()
                                .Contains("profiler")
                                ?? false,
```